### PR TITLE
Eliminate a heap allocation from PartialMerkleTree encoding & decoding

### PR DESCRIPTION
**Pull request by stevenroose**
Just came across this and felt like doing this.

### CODE REVIEW

**Comment by apoelstra** [comment]
Nice! This is a pretty cool use for `chunks` (which I confirmed will yield the remainder if you don't have a multiple-of-8 number of bits).

**Review by apoelstra — APPROVED** [review]
ACK fa10668a353fbfff4625e7bef18c9e6712322b4e

**Review by tcharding — APPROVED** [review]
ACK fa10668a353fbfff4625e7bef18c9e6712322b4e

File: bitcoin/src/merkle_tree/block.rs [hunk 1]
```diff
@@ -432,26 +432,38 @@ impl PartialMerkleTree {
 
 impl Encodable for PartialMerkleTree {
     fn consensus_encode<W: io::Write + ?Sized>(&self, w: &mut W) -> Result<usize, io::Error> {
-        let ret = self.num_transactions.consensus_encode(w)? + self.hashes.consensus_encode(w)?;
-        let mut bytes: Vec<u8> = vec![0; (self.bits.len() + 7) / 8];
```
Comment (yancyribbens) [inline bitcoin/src/merkle_tree/block.rs:436]: Nice!

**Pull request merged**